backend/app/synthesize_polygons/synthesize_polygon.py

```diff
@@ -83,15 +86,32 @@ def sides_of_polygon(points):
     return side_lengths
 
 
-# for future feature
-# def durations_from_sides(sides, base_duration):
-#     """
-#     Comptue the duration of each side based on the ratio to the first side and the base duration.
-#     :param sides: a list of the side lengths
-#     :param base_duration: the duration (in seconds) of the first side
-#     :return: a list of durations of each side
-#     """
-#     return [(side/side[0])*base_duration for side in sides]
+def sides_to_duration(sides, base_duration):
+    """
+    Compute the duration of each side based on the ratio to the first side and the base duration.
+    :param sides: a list of the side lengths
+    :param base_duration: the duration (in seconds) of the first side
+    :return: a tuple consisting of a list of durations of each side as well as the total length
+    of the sound as an integer
+    """
+    duration_list = [(side/sides[0])*base_duration for side in sides]
+    total_duration = 0
+    for duration in duration_list:
+        total_duration += duration
+    return duration_list, total_duration
+
+
+def amplitude_decay(frequency):
+    """
+    Decays the amplitude to counterbalance psycho-acoustic effects in which we hear higher
+    pitches as louder. Returns the decay as a scaling factor for amplitude.
+    :param frequency: frequency of the note in Hz
+    :return: an amplitude scaling factor
+    """
+    if frequency < 700:
+        return 1
+    # exponential decay based on frequency
+    return 0.5 ** ((frequency - 700) / 2000)
```

Comment:
There are some exponential decay / growth functions over in one of the synthesis functions: probably worth factoring out and generalizing. They tend to be useful in audio apps.